azurerm/resource_arm_storage_account.go

```diff
@@ -1484,6 +1568,27 @@ func flattenStorageAccountVirtualNetworks(input *[]storage.VirtualNetworkRule) [
 	return virtualNetworks
 }
 
+func flattenBlobProperties(input storage.BlobServiceProperties) []interface{} {
+	if input.BlobServicePropertiesProperties == nil {
+		return []interface{}{}
+	}
+
+	blobProperties := make(map[string]interface{})
+
+	if deletePolicy := input.BlobServicePropertiesProperties.DeleteRetentionPolicy; deletePolicy != nil {
+		if enabled := deletePolicy.Enabled; enabled != nil && *enabled {
+			deleteRetentionPolicy := make([]interface{}, 0)
+			attr := make(map[string]interface{})
+
+			attr["days"] = int(*deletePolicy.Days)
+
+			blobProperties["delete_retention_policy"] = append(deleteRetentionPolicy, attr)
```

Comment:
given there's always a value for all fields (even if a `nil` response is returned from the API - we can always set this field into the nested object via:

```suggestion
			deleteRetentionPolicies = append(deleteRetentionPolicies, map[string]interface{}{
				"days": days,
			})
```

Reply:
done